contributor-docs/adrs/adr-008-experimental-components.md

```diff
@@ -0,0 +1,70 @@
+# ADR 008: Lowering the barrier to entry for experimental components
+
+## Status
+
+Proposed
+
+&nbsp;
+
+## Context
+
+#### Recap: Drafts
+
+As we work on new components (or rewrites of old components), we often need to start with a "prototype"[^1] and not include them in the semantically versioned (semver-ed) main bundle.
+
+The approach that has served us well since Dec 2021 has been "drafts" (along with causing a lot of confusion).
+
+[ADR-006](./adr-006-drafts.md) explains our use of drafts, but here are the relevant parts:
+
+- When we are not sure about the API of a component, we put it in a different bundle: "primer/react/drafts"
+- A draft has experimental API, but follows the development conventions of primer/react
+- Components in drafts are accompanied by a design spec and go through the process of accessibility reviews. A component can check multiple boxes in the component lifecycle.
+- The contract with consumers is that you are opting into a component in which we don't have 100% confidence yet.
+
+[^1]: Calling this a prototype and not experimental, because that already means something else in the maturity lifecycle.
+
+&nbsp;
+
+#### Problems with drafts
+
+1. **Incompatible with semver**: Because the expectation with drafts is that we are not 100% confident about the API or structure of a component, we slip in breaking changes to draft components as minor releases to primer/react. On the consumer side, this makes every upgrade, even a patch, a potentially breaking upgrade. So, you are effectively opting out of your semver.
+
+1. **Confusion about maturity**: Because of the generic name, "drafts" causes a lot of confusion, especially with "experimental". It is a prototype or experiment, but NOT in the way the [component lifecycle]([https://primer.style/contribute/component-lifecycle]) defines "Experimental" maturity.
+
+1. **Need for experimental candidates**: As more teams build React components and look to upstream reusable components, it is natural to look at drafts as the place to put them. However, the barrier to entry into drafts isn't low (because we still follow design reviews, dev conventions, API reviews, etc.)
+
+Upstreaming a component that was built in the context of a product requires re-evaluating the baked-in assumptions and most likely a change in the API to make it useful for other contexts.
+
+&nbsp;
+
+## Decision
+
+1. "Experimental" components, as defined by the component lifecycle, should not be part of semantically versioned npm package for `@primer/react`.
```

Comment:
Would the "Experimental" components become the same as the current "Drafts"? or will they still be "Experimental" but live outside of Primer?

Reply:
> If we move "experimental" components out of @primer/react into a new package, would we stop using the drafts directory?

yes. we would only move them to `@primer/react` package once we can commit to non-breaking changes

> Would the "Experimental" components become the same as the current "Drafts"? or will they still be "Experimental" but live outside of Primer?

I'd like to detach them from current "drafts" because "draft components" go through the primer workflow of design, API, sometimes a11y review. "Experimental" components should have a lower barrier to entry.

From: https://github.com/primer/react/pull/2214/files#diff-5e30a807664dafbee9f227901a2459809a2c2226e183880f8bf6e05b1f52a7ceR20-R21:

> - A draft has experimental API, but follows the development conventions of primer/react
> - Components in drafts are accompanied by a design spec and go through the process of accessibility reviews. A component can check multiple boxes in the component lifecycle.

Reply:
Maybe we update the expectation for drafts components esp regards to. Perhaps add an ETA for drafts to move to main component directory. Have a pre-alpha checklist that does not include accessibility review?

Reply:
> Maybe we update the expectation for drafts components esp regards to. Perhaps add an ETA for drafts to move to main component directory.

Can you say more about this, who is the ETA for?